notebooks/autodiff_cookbook.ipynb

```diff
@@ -0,0 +1,1062 @@
+        "print(grad(grad(grad(np.tanh)))(2.0))"
+      ],
+      "execution_count": 5,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "-0.13621867\n",
+            "0.25265405\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "id": "2rcnpTiinqi8",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Let's look at computing gradients with `grad` in a linear logistic regression model. First, the setup:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "27TcOT2i6vu5",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def sigmoid(x):\n",
+        "  return 0.5 * (np.tanh(x / 2) + 1)\n",
+        "\n",
+        "# Outputs probability of a label being true.\n",
+        "def predict(W, b, inputs):\n",
+        "  return sigmoid(np.dot(inputs, W) + b)\n",
+        "\n",
+        "# Build a toy dataset.\n",
+        "inputs = np.array([[0.52, 1.12,  0.77],\n",
+        "                   [0.88, -1.08, 0.15],\n",
+        "                   [0.52, 0.06, -1.30],\n",
+        "                   [0.74, -2.49, 1.39]])\n",
+        "targets = np.array([True, True, False, True])\n",
+        "\n",
+        "# Training loss is the negative log-likelihood of the training examples.\n",
+        "def loss(W, b):\n",
+        "  preds = predict(W, b, inputs)\n",
+        "  label_probs = preds * targets + (1 - preds) * (1 - targets)\n",
+        "  return -np.sum(np.log(label_probs))\n",
+        "\n",
+        "# Initialize random model coefficients\n",
+        "key, W_key, b_key = random.split(key, 3)\n",
+        "W = random.normal(W_key, (3,))\n",
+        "b = random.normal(b_key, ())"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "8Wk-Yai7ooh1",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Use the `grad` function with its `argnums` argument to differentiate a function with respect to positional arguments."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "bpmd8W8-6vu6",
+        "colab_type": "code",
+        "outputId": "04affbcf-0e92-4045-f09e-ee43aba4b306",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 104
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "# Differentiate `loss` with respect to the first positional argument:\n",
+        "W_grad = grad(loss, argnums=0)(W, b)\n",
+        "print('W_grad', W_grad)\n",
+        "\n",
+        "# Since argnums=0 is the default, this does the same thing:\n",
+        "W_grad = grad(loss)(W, b)\n",
+        "print('W_grad', W_grad)\n",
+        "\n",
+        "# But we can choose different values too:\n",
+        "b_grad = grad(loss, 1)(W, b)\n",
```

Comment:
You might wish to always pass `argnums` by name here (and below). Consistency is a good thing. If you wish to switch to positional, perhaps comment that you are doing so.

Reply:
Added a comment.